apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs

```diff
@@ -444,6 +444,13 @@ public override bool NextResult()
                             {
                                 throw new SpannerException(ErrorCode.DeadlineExceeded, "Read operation timed out");
                             }
+                            // This excpetion may occur if Spanner had to split returned data into multiple PartialResultSets.
+                            // This should not happen right now as DML statement can never execute more than 40,000 mutations but could increase in the future.
+                            catch (Exception ex) when (ex.Message.Contains("Read/query was cancelled due to Commit()/Rollback() on the enclosing transaction"))
```

Comment:
Hi @jskeet  and @olavloite , Please let me know that does this seems correct way to handle the case if the result was so big that Spanner had to split it into multiple PartialResultSets. If not how and where do you suggest us to do this error handling. Also what I understood was that right now this error is not reproducible , am I correct in saying so ?

Reply:
We should *definitely* only catch the right exception type - which I'd expect to be `RpcException`.
Ideally Spanner will provide an `ErrorInfo` that we can use to check the condition rather than checking text.

Reply:
When we've heard back about this, we should have a test for it as well.

Reply:
(Sorry, I have not found the time to give this a complete review)

It is possible to reproduce it, but not with a DML statement. What you can do is the following:

1. Create a table with a large number of rows/columns (basically enough data that Spanner will return multiple PartialResultSets)
2. Start a read/write transaction and execute `select * from my_table` using that transaction. Make sure to call `DataReader.Read` once while the transaction is still active.
3. Call `Transaction.Commit()`.
4. Try to fetch the rest of the data from the reader. This will fail with the above-mentioned error.

I don't think Spanner includes any unique `ErrorInfo` with this error, so I'm afraid that we have to rely on text matching.